### PR TITLE
Add compatibility with lxc 3

### PR DESCRIPTION
**Pull request by ken2812221**
There are some options that have been renamed in lxc 3.

### CODE REVIEW

**Comment by ken2812221** [comment]
```diff
diff --git a/etc/lxc.config.in b/etc/lxc.config.in
index 8879f43..04ebb19 100644
--- a/etc/lxc.config.in
+++ b/etc/lxc.config.in
@@ -1,6 +1,6 @@
-lxc.tty = 4
-lxc.pts = 1024
-lxc.rootfs = ROOTFS
+lxc.tty.max = 4
+lxc.pty.max = 1024
+lxc.rootfs.path = ROOTFS
 lxc.arch = ARCH
 lxc.cgroup.devices.deny = a
 # /dev/null and zero
@@ -28,10 +28,10 @@ lxc.mount.entry=sysfs ROOTFS/sys sysfs defaults  0 0
 # On the host, run: ifconfig br0 up 10.0.2.2
 # Alternatively, you can use another IP range for the bridge interface, in this case set
 # the environment variables GITIAN_HOST_IP and LXC_GUEST_IP appropriately.
-lxc.network.type = veth
-lxc.network.flags = up
-lxc.network.link = GUESTLINK
-lxc.network.ipv4 = GUESTIP/24
-lxc.network.ipv4.gateway = auto
+lxc.net.0.type = veth
+lxc.net.0.flags = up
+lxc.net.0.link = GUESTLINK
+lxc.net.0.ipv4.address = GUESTIP/24
+lxc.net.0.ipv4.gateway = auto

-lxc.utsname = gitian
+lxc.uts.name = gitian

**Comment by devrandom** [comment]
Thank you for the contribution.  What distro are you testing this on?

**Comment by ken2812221** [comment]
I tested it on Ubuntu 18.04.

File: libexec/config-lxc [hunk 1]
```diff
@@ -10,4 +10,10 @@ if [ -z "$LXC_BRIDGE" ]; then
     LXC_BRIDGE=br0
 fi
 
-sed "s;ROOTFS;$wd/target-$LXC_SUITE-$LXC_ARCH;;s;ARCH;$LXC_ARCH;g;;s;GUESTIP;$LXC_GUEST_IP;g;s;GUESTLINK;$LXC_BRIDGE;g" < etc/lxc.config.in > var/lxc.config
+if [[ $(lxc --version) = 3.* ]]; then
```
Comment (achow101) [inline libexec/config-lxc]: IMO this should be the other way around. if the version is 2 or earlier, it should use the old config. The new config should be the default as that is what will be used going forwards.

Comment (achow101) [reply libexec/config-lxc]: Instead of using `lxc --version`, I think you should use `lxc-start --version`.

**Comment by achow101** [comment]
Concept ACK.

I am testing this now. I ran into an issue where the `lxc` binary does not come with Ubuntu's `lxc` package so I had to change the version check to use `lxc-start`. I think it is safer to do that as we also use `lxc-start` elsewhere for a version check and that command shouldn't be going away any time soon.

**Comment by devrandom** [comment]
Verified @achow101's point that `lxc` doesn't actually exist after `apt install lxc`, but `lxc-start` does.

**Comment by devrandom** [comment]
ACK otherwise

File: libexec/config-lxc (hunk 1 above)
Comment (ken2812221) [reply libexec/config-lxc]: Fixed

**Pull request merged**